wallet/src/de/schildbach/wallet/util/TimeUtils.kt

```diff
@@ -42,30 +44,49 @@ private fun queryNtpTime(server: String): Long? {
     return null
 }
 
+@Throws(NullPointerException::class)
 suspend fun getTimeSkew(): Long {
     var networkTime: Long? = null
-    try {
-        networkTime = queryNtpTime("pool.ntp.org")
-    } catch (e: SocketTimeoutException) {
-        // swallow, the next block will use alternate method
+    var timeSource = "NTP"
+
+    val networkTimes = arrayListOf<Long>()
+    for (i in 0..3) {
+        try {
+            val time = queryNtpTime("pool.ntp.org")
+            if (time != null && time > 0) { networkTimes.add(time) }
+        } catch (e: SocketTimeoutException) {
+            // swallow
+        }
+    }
+    networkTimes.sort()
+    when (networkTimes.size) {
+        3 -> networkTime = networkTimes[2]
+        2 -> networkTime = (networkTimes[0] + networkTimes[1]) / 2
+        else -> { }
     }
```

Comment:
perhaps later we should use the same mechanism of Dash Core for timeskew, as this gets complicated.

Sometimes the NTP server returns a number less than zero, so this is ignored.

Sometimes when the app starts up, there is a dialog about 6 or 7 second timeskew, but subsequent calls return 1 or 2 seconds difference.